lib/Parse/ParseDecl.cpp

```diff
@@ -6348,21 +6371,7 @@ void Parser::parseAbstractFunctionBody(AbstractFunctionDecl *AFD) {
 
   Scope S(this, ScopeKind::FunctionBody);
 
-  // Enter the arguments for the function into a new function-body scope.  We
-  // need this even if there is no function body to detect argument name
-  // duplication.
-  if (auto *P = AFD->getImplicitSelfDecl())
-    addToScope(P);
-  addParametersToScope(AFD->getParameters());
-
-   // Establish the new context.
-  ParseFunctionBody CC(*this, AFD);
```

Comment:
This changes the lifetime of the body context.  Is that correct? Is there any risk this will cause issues if someone modifies this in the future?  The body is not "finished" here, since we have the transform below.

Reply:
The transformation below doesn't use `CurDeclContext` nor `CurDeclContext`. So this lifetime change should not affect it. Am I missing something?

> Is there any risk this will cause issues if someone modifies this in the future? 

In general, a parser developer should know what the current `CurDeclContext`. What is your concern exactly?

Reply:
Okay, just checking.